src/meltano/core/tracking/tracker.py

```diff
@@ -100,17 +118,80 @@ def __init__(
         else:
             self.snowplow_tracker = None
 
-        self.send_anonymous_usage_stats = self.settings_service.get(
-            "send_anonymous_usage_stats", True
-        )
+        stored_telemetry_settings = self.load_saved_telemetry_settings()
+        self.client_id = stored_telemetry_settings.client_id or uuid.uuid4()
 
-        project_ctx = ProjectContext(project)
-        self.project_id = str(project_ctx.project_uuid)
+        project_ctx = ProjectContext(project, self.client_id)
+        self.project_id: uuid.UUID = project_ctx.project_uuid
         self.contexts: tuple[SelfDescribingJson] = (
             environment_context,
             project_ctx,
         )
 
+        self.telemetry_state_change_check(stored_telemetry_settings)
+
+    @cached_property
+    def send_anonymous_usage_stats(self) -> bool:
+        """Return whether anonymous usages stats are enabled (bool).
+
+        - Return the value from 'send_anonymous_usage_stats', if set.
+        - Otherwise the opposite of 'tracking_disabled', if set.
+        - Otherwise return 'True'
+
+        Returns:
+            True if anonymous usage stats are enabled.
+        """
+        if self.settings_service.get("send_anonymous_usage_stats", None) is not None:
+            return self.settings_service.get("send_anonymous_usage_stats")
+
+        if self.settings_service.get("tracking_disabled", None) is not None:
+            return not self.settings_service.get("tracking_disabled")
+
+        return True
```

Comment:
Yes, this is the bit that resolves the removal of the `env_alias` `MELTANO_TRACKING_DISABLED`.